lambda_functions/va_profile/va_profile_opt_in_out_lambda.py

```diff
@@ -124,23 +155,26 @@ def va_profile_opt_in_out_lambda_handler(event: dict, context) -> dict:
 
         try:
             params = (                             # Stored function parameters:
-                record["VaProfileId"],             #     _va_profile_id
-                record["CommunicationItemId"],     #     _communication_item_id
-                record["CommunicationChannelId"],  #     _communication_channel_name
+                record["vaProfileId"],             #     _va_profile_id
+                record["communicationItemId"],     #     _communication_item_id
+                record["communicationChannelId"],  #     _communication_channel_name
                 record["allowed"],                 #     _allowed
                 record["sourceDate"],              #     _source_datetime
             )
 
             if db_connection is None or db_connection.status != 0:
                 # Attempt to (re-)establish a database connection
-                db_connection = make_connection()
+                db_connection = make_connection(worker_id)
 
             if db_connection is None:
                 raise RuntimeError("No database connection.")
 
             # Execute the stored function.
             with db_connection.cursor() as c:
-                put_record["status"] = "COMPLETED_SUCCESS" if c.execute(OPT_IN_OUT_QUERY, params) else "COMPLETED_NOOP"
+                # https://www.psycopg.org/docs/cursor.html#cursor.execute
+                c.execute(OPT_IN_OUT_QUERY, params)
+                put_record["status"] = "COMPLETED_SUCCESS" if c.fetchone()[0] else "COMPLETED_NOOP"
+                db_connection.commit()
```

Comment:
In the exception handling, there is a setting of response["statusCode"]. Since the put_record["status"] for both 400/500 responses is "Completed_Failure" there would be no way to determine which exception occurred so a key error further down the loop would overwrite a general exception that occurred earlier.  We may want to consider making statuses for completed internal failure and completed key error to be more specific about errors encountered.